controller/adr.rst

```diff
@@ -0,0 +1,189 @@
+
+.. configuration-block::
+
+    .. code-block:: php-annotations
+
+        # src/AppBundle/Action/HelloAction.php
+        // ...
+
+        /**
+         * @Route("/hello", name="hello")
+         */
+        final class HelloAction
+        {
+            // ...
+        }
+
+    .. code-block:: yaml
+
+        # app/config/routing.yml
+        hello:
+            path:     /hello
+            defaults: { _controller: AppBundle\Action\HelloAction }
+
+    .. code-block:: xml
+
+        <!-- app/config/routing.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <routes xmlns="http://symfony.com/schema/routing"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/routing
+                http://symfony.com/schema/routing/routing-1.0.xsd">
+
+            <route id="hello" path="/hello">
+                <default key="_controller">AppBundle\Action\HelloAction</default>
+            </route>
+
+        </routes>
+
+    .. code-block:: php
+
+        // app/config/routing.php
+        use AppBundle\Action\HelloAction
+        
+        $collection->add('hello', new Route('/hello', array(
+            '_controller' => HelloAction::class,
+        )));
+
+Accessing the request
+---------------------
+
+As you can imagine, as the logic evolve, your class is capable of accessing
+the request from simple method injection like this ::
+
+    <?php
+
+        // ...
+
+        public function __invoke(Request $request)
+        {
+            return new Response($this->twig->render('default/index.html.twig'));
+        }
+    }
+
+Like you can easily imagine, the container is the best option to gain access to ths request,
+using this approach, a simple update is recommended ::
+
+    <?php
+
+    namespace AppBundle\Action;
+
+    use Twig\Environment;
+    use Symfony\Component\HttpFoundation\Request;
+    use Symfony\Component\HttpFoundation\RequestStack;
+    use Symfony\Component\Templating\EngineInterface;
+
+    final class HelloAction
+    {
+        private $requestStack;
+
+        private $twig;
+
+        public function __construct(RequestStack $requestStack, Environment $twig)
+        {
+            $this->requestStack = $requestStack
+            $this->twig = $twig;
+        }
+
+        public function __invoke(Request $request)
+        {
+            return new Response($this->twig->render('default/index.html.twig'));
+        }
+    }
+
+This way, you can easily access to parameters::
+
+    <?php
+
+        // ...
+
+        public function __construct(RequestStack $requestStack, Environment $twig)
+        {
+            $this->requestStack = $requestStack
+            $this->twig = $twig;
+        }
+
+        public function __invoke(Request $request)
+        {
+            $data = $this->requestStack->getCurrentRequest()->get('id');
+
+            return new Response($this->twig->render('default/index.html.twig', ['data' => $data]));
```

Comment:
We still use the long syntax in the core and in the docs to ease merging olders branches.